components/data_prefetch/data_prefetch.test.tsx

```diff
@@ -66,124 +72,212 @@ describe('/components/data_prefetch', () => {
         })],
     };
 
-    test('should call posts of current channel when it is set', async () => {
+    beforeEach(() => {
+        mockQueue.splice(0, mockQueue.length);
+    });
+
+    test('should fetch posts for current channel on first channel load', async () => {
+        const props = defaultProps;
         const wrapper = shallow<DataPrefetch>(
-            <DataPrefetch {...defaultProps}/>,
+            <DataPrefetch {...props}/>,
         );
-        const instance = wrapper.instance();
 
+        const instance = wrapper.instance();
         instance.prefetchPosts = jest.fn();
+
+        // Change channels and wait for async componentDidUpdate to resolve
         wrapper.setProps({currentChannelId: 'currentChannelId'});
+        await Promise.resolve(true);
+
+        expect(mockQueue).toHaveLength(1);
+        expect(instance.prefetchPosts).not.toHaveBeenCalled();
+
+        // Manually run queued tasks
+        mockQueue.shift()!();
+        expect(instance.prefetchPosts).toHaveBeenCalledTimes(1);
         expect(instance.prefetchPosts).toHaveBeenCalledWith('currentChannelId');
-        await loadProfilesForSidebar();
-        expect(defaultProps.actions.trackDMGMOpenChannels).toHaveBeenCalled();
     });
 
-    test('should call for LHS profiles and also call for posts based on prefetchQueueObj', async () => {
+    test('should track opened DMs/GMs on first channel load', async () => {
+        const props = defaultProps;
         const wrapper = shallow<DataPrefetch>(
-            <DataPrefetch {...defaultProps}/>,
+            <DataPrefetch {...props}/>,
         );
-        const instance = wrapper.instance();
-        instance.prefetchPosts = jest.fn();
+
+        expect(props.actions.trackDMGMOpenChannels).not.toHaveBeenCalled();
+
+        // Change channels
+        wrapper.setProps({currentChannelId: 'currentChannelId'});
+        await Promise.resolve(true);
```

Comment:
I'm guessing there's no other way of waiting for the component to update and ensure that the function has been called?

Reply:
The component is updating fine, but its `componentDidUpdate` only runs up until it awaits a call ([link](https://github.com/mattermost/mattermost-webapp/blob/master/components/data_prefetch/data_prefetch.tsx#L58)). After that, it waits for the test to pause again for its chance to run. When the test awaits on anything at all, it lets `componentDidUpdate` continue running. The previous version awaited random mocked functions, so I replaced those with dummy promises to be clearer that it's just to let other code proceed.